edx_django_utils/monitoring/internal/middleware.py

```diff
@@ -26,6 +27,58 @@
 _REQUEST_CACHE_NAMESPACE = f'{_DEFAULT_NAMESPACE}.custom_attributes'
 
 
+class DeploymentMonitoringMiddleware:
+    """
+    Middleware to record environment values at the time of deployment for each service.
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        self.record_python_version()
+        self.record_django_version()
+        response = self.get_response(request)
+        return response
+
+    @staticmethod
+    def record_django_version():
+        """
+        Record the installed Django version as custom attribute
+
+        .. custom_attribute_name: django_version
+        .. custom_attribute_description: The django version in use (e.g. '2.2.24').
+           Set by DeploymentMonitoringMiddleware.
+        """
+        if not newrelic:  # pragma: no cover
+            return
+        try:
+            import django  # pylint: disable=import-outside-toplevel
+            django_version = django.__version__
+        except ModuleNotFoundError:
+            return
+        _set_custom_attribute('django_version', django_version)
+
+    @staticmethod
+    def record_python_version():
+        """
+        Record the Python version as custom attribute
+
+        .. custom_attribute_name: python_version
+        .. custom_attribute_description: The Python version in use (e.g. '3.8.10').
+           Set by DeploymentMonitoringMiddleware.
+        """
+        if not newrelic:  # pragma: no cover
+            return
+        _set_custom_attribute('python_version', platform.python_version())
+
+    def process_exception(self, exception):  # pylint: disable=unused-argument
```

Comment:
I suspect this isn’t needed and will just add extra processing for errors. Can we leave this out and add it in only if I am wrong?

Reply:
@UsamaSadiq: Reminder to kill this, or at least respond to this comment. Thanks.

Reply:
I've removed this method as well.